src/provider.ts

```diff
@@ -0,0 +1,22 @@
+import {ethers} from "ethers";
+import {EthereumNetwork} from "./types";
+
+let savedWeb3Provider = ethers.getDefaultProvider(EthereumNetwork.Default);
+let savedWallet: ethers.Wallet | undefined;
```

Comment:
can just import {Wallet , providers} from "ethers". easy to write type everywhere.